src/basic/socket-util.c

```diff
@@ -237,7 +237,9 @@ int socket_address_parse_and_warn(SocketAddress *a, const char *s) {
 }
 
 int socket_address_parse_netlink(SocketAddress *a, const char *s) {
-        int family;
+        int family = 0; /* Initalize to work around false positive in msan:
+                         * https://github.com/google/sanitizers/issues/767.
+                         * This can be removed once the issue is fixed. */
```

Comment:
This doesn't seem to get around the bug. `test-unit-file` is still failing with
```sh
Uninitialized bytes in __interceptor_strcmp at offset 0 inside [0x707000000000, 1)
==5==WARNING: MemorySanitizer: use-of-uninitialized-value
    #0 0x7f0bd1a280ce in strcmp_ptr /work/build/../../src/systemd/src/basic/string-util.c:41:24
    #1 0x7f0bd1a1deda in streq_ptr /work/build/../../src/systemd/src/basic/string-util.h:51:16
    #2 0x7f0bd1a17cec in netlink_family_from_string /work/build/../../src/systemd/src/basic/socket-util.c:799:1
    #3 0x7f0bd1a17899 in socket_address_parse_netlink /work/build/../../src/systemd/src/basic/socket-util.c:255:18
    #4 0x4bc2d8 in config_parse_socket_listen /work/build/../../src/systemd/src/core/load-fragment.c:393:21
    #5 0x7fb23a in next_assignment /work/build/../../src/systemd/src/shared/conf-parser.c:155:32
    #6 0x7ef30a in parse_line /work/build/../../src/systemd/src/shared/conf-parser.c:273:16
    #7 0x7ed1fa in config_parse /work/build/../../src/systemd/src/shared/conf-parser.c:390:21
    #8 0x4b2f78 in LLVMFuzzerTestOneInput /work/build/../../src/systemd/src/fuzz/fuzz-unit-file.c:48:16
    #9 0x8a9d96 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) /src/libfuzzer/FuzzerLoop.cpp:515:13
    #10 0x864221 in fuzzer::RunOneTest(fuzzer::Fuzzer*, char const*, unsigned long) /src/libfuzzer/FuzzerDriver.cpp:280:6
    #11 0x878476 in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) /src/libfuzzer/FuzzerDriver.cpp:703:9
    #12 0x863361 in main /src/libfuzzer/FuzzerMain.cpp:20:10
    #13 0x7f0bd054682f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)
    #14 0x4342b8 in _start (/out/fuzz-unit-file+0x4342b8)

  Uninitialized value was created by a heap allocation
    #0 0x465990 in __interceptor_malloc /src/llvm/projects/compiler-rt/lib/msan/msan_interceptors.cc:939
    #1 0x7f0bd0583599 in _IO_vfscanf (/lib/x86_64-linux-gnu/libc.so.6+0x5d599)
    #2 0x7f0bd0596276 in vsscanf (/lib/x86_64-linux-gnu/libc.so.6+0x70276)
```
. The only way to get around the issue I was able to think of is to stop using `%ms` and manually allocate memory and pass the maximum field width to `sscanf`.

Basically, I don't think it is possible to get around every bug in sanitizers and this should be done only in the last resort. I think it would be better to try to file a bug https://github.com/google/oss-fuzz/issues/new first as suggested in https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=6884:
> If this is incorrect, please file a bug on https://github.com/google/oss-fuzz/issues/new

Reply:
I opened https://github.com/google/oss-fuzz/issues/1247. I hope there will be a way to ignore the failure until https://github.com/google/sanitizers/issues/767 is fixed

Reply:
This might work:
https://clang.llvm.org/docs/MemorySanitizer.html#attribute-no-sanitize-memory

Haven't tested it...

Reply:
I recently came around another false positive which haven't been reported by `OSS-Fuzz` yet and I'm sure these two aren't the last ones. I don't think the code should be patched just to make `msan` happy.